src/main/java/org/testng/TestNG.java

```diff
@@ -219,6 +223,22 @@ public void setOutputDirectory(final String outputdir) {
     }
   }
 
+  /**
+   * Sets the possibility to generate TestResult attributes
+   * @param generateTestResultAttributes true / false.
+   */
+  public void setGenerateTestResultAttributes(final Boolean generateTestResultAttributes) {
+    m_generateTestResultAttributes = generateTestResultAttributes;
```

Comment:
What does it mean if the value is null here and below?

Reply:
it can't be null. I use Boolean instead of boolean.

Reply:
Exactly: `Boolean` is nullable, `boolean` is not.
So `Boolean` is true, false or null.

Reply:
Oh, you are right. I add verification for null